src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -0,0 +1,124 @@
+    climberMotorBack = new TalonFX(MotorIDConstants.kCLimberMotorFollower);
+    climberConfigsToClimber =
+        new Slot0Configs()
+            .withKP(ClimberConstants.kClimberP0)
+            .withKI(ClimberConstants.kClimberI0)
+            .withKD(ClimberConstants.kClimberD0)
+            .withKG(ClimberConstants.kClimberkG0)
+            .withKV(ClimberConstants.kClimberkV0);
+    climberConfigsAtClimber =
+        new Slot1Configs()
+            .withKP(ClimberConstants.kClimberP1)
+            .withKI(ClimberConstants.kClimberI1)
+            .withKD(ClimberConstants.kClimberD1)
+            .withKG(ClimberConstants.kClimberkG1)
+            .withKV(ClimberConstants.kClimberkV1);
+    climberMotorFront.getConfigurator().apply(climberConfigsToClimber);
+    climberMotorBack.getConfigurator().apply(climberConfigsToClimber);
+
+    if (Robot.isSimulation()) {
+      simClimberGearbox = DCMotor.getKrakenX60(1);
+      climberSim =
+          new SingleJointedArmSim(simClimberGearbox, 126, 0.077, 0.1, 0, Math.PI, false, 1);
+      climbMech = new Mechanism2d(2, 2);
+      climbMechRoot = climbMech.getRoot("Climb Mech root", 1, 1);
+      climbMechLigament =
+          climbMechRoot.append(
+              new MechanismLigament2d(
+                  "Climb Mech Ligament", 1, Radians.of(climberSim.getAngleRads()).in(Degrees)));
+      SmartDashboard.putData("Climb Mech", climbMech);
+    }
+  }
+
+  private Command runClimber(Angle position) {
+    return runOnce(
+        () -> {
+          climberMotorFront.setControl(
+              new PositionDutyCycle(position.times(ClimberConstants.KGearRatio)));
+          climberMotorBack.setControl(
+              new PositionDutyCycle(position.times(ClimberConstants.KGearRatio * -1)));
+        });
+  }
+
+  private BooleanSupplier isClimberAtPosition(Angle position) {
+    return () ->
+        Math.abs((climberMotorFront.getPosition().getValue().in(Degrees) - position.in(Degrees)))
+            < 5;
+  }
+
+  public Command climb() {
+    return Commands.sequence(
+        runClimber(ClimberConstants.kClimberCage),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberCage)),
+        runClimber(ClimberConstants.kClimberExtended),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberExtended)));
+  }
+
+  public Command retract() {
+    return Commands.sequence(
+        runClimber(ClimberConstants.kClimberCage),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberCage)),
+        runClimber(ClimberConstants.kClimberRetracted),
+        Commands.waitUntil(isClimberAtPosition(ClimberConstants.kClimberRetracted)));
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    if (Robot.isSimulation()) {
+      climberSim.setInput(climberMotorFront.get());
+      climberSim.update(0.02);
+      climbMechLigament.setAngle(Radians.of(climberSim.getAngleRads()).in(Degrees));
```

Comment:
You are missing setting the position feedback to the motor controller. Something like this:
`climberMotorFront.setPosition(Radians.of(climberSim.getAngleRads()));`